src/plugins/intel_gpu/tests/test_cases/concatenation_gpu_test.cpp

```diff
@@ -856,4 +1011,163 @@ TEST(concat_gpu_onednn, basic_input_types) {
+                auto offset = weights_lay.get_linear_offset(coords);
+                weights_ptr[offset] = static_cast<Type>(1.f);
+            }
+        }
+        topology.add(data("weights" , weights_mem));
+        topology.add(convolution("conv", "concat", { "weights" }));
+        topology.add(pooling("pool_final", "conv", pooling_mode::max, {1, 1, 1, 1}, {1, 1, 1, 1}));
+        topology.add(reorder("reorder", "pool_final", layout(data_type, format::bfyx, {batch_num, output_f, input_y, input_x})));
+
+        network concat_network(engine, topology, options);
+        for (size_t i = 0; i < in_features.size(); i++) {
+            concat_network.set_input_data(input_ids[i], in_memory[i]);
+        }
+        concat_network.execute();
+
+        for (auto i : concat_network.get_primitives_info()) {
+            // std::cout << "  " << i.original_id << " " << i.kernel_id << std::endl;
+            if (i.original_id == "concat") {
+                if (options.get<build_option_type::optimize_data>()->enabled()) {
+                    EXPECT_TRUE(i.kernel_id == "undef");
+                } else {
+                    EXPECT_FALSE(i.kernel_id == "undef");
+                }
+            }
+        }
+
+        return concat_network.get_output("reorder").get_memory();
+    }
+
+    std::vector<std::vector<std::vector<std::vector<std::vector<Type>>>>> generate_input() {
+        const size_t batch_num = testing::get<0>(GetParam());
+        const std::vector<size_t> in_features = testing::get<1>(GetParam());
+        const size_t input_y = testing::get<2>(GetParam());
+        const size_t input_x = testing::get<3>(GetParam());
+
+        std::vector<std::vector<std::vector<std::vector<std::vector<Type>>>>> input(in_features.size());
+        for (size_t i = 0; i < in_features.size(); ++i) {
+            input[i] = generate_random_4d<Type>(batch_num, in_features[i], input_y, input_x, -1, 1);
+        }
+        return input;
+    }
+
+    void test(format::type fmt) {
+        auto input = generate_input();
+
+        // implicit concat
+        build_options options1;
+        options1.set_option(build_option::optimize_data(true));
+        implementation_desc impl = { fmt, std::string(""), impl_types::onednn };
+        options1.set_option(build_option::force_implementations({ {"conv", impl} }));
+        auto out_mem1 = run_concat_network(input, fmt, options1);
+        cldnn::mem_lock<Type> out_ptr1(out_mem1, get_test_stream());
+
+        // explicit concat
+        build_options options2;
+        options2.set_option(build_option::optimize_data(false));
+        auto out_mem2 = run_concat_network(input, fmt, options2);
+        cldnn::mem_lock<Type> out_ptr2(out_mem2, get_test_stream());
+
+        EXPECT_EQ(out_ptr1.size(), out_ptr2.size());
+        size_t diff_count = 0;
+        for (size_t i = 0; i < out_ptr1.size(); ++i) {
+            if (out_ptr1[i] != out_ptr2[i]) diff_count++;
+        }
+        EXPECT_EQ(diff_count, 0);
```

Comment:
I think we need to confirm that implicit concat is working well as expected. That is, check whether concat is optimized out or not.